iota-trinary/src/trytes.rs

```diff
@@ -0,0 +1,675 @@
+    /// buffer.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `idx` is larger than the `String`'s length.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use iota_trinary::*;
+    ///
+    /// let mut t = Trytes::try_from("BAR").unwrap();
+    ///
+    /// t.insert_str(0, "FOO").unwrap();
+    ///
+    /// assert_eq!("FOOBAR", t.as_str());
+    /// ```
+    pub fn insert_str(&mut self, idx: usize, string: &str) -> Result<()> {
+        let bytes = string.as_bytes();
+        Self::all_tryte_alphabete(bytes.iter().copied())?;
+        unsafe {
+            self.insert_bytes(idx, bytes);
+        }
+        Ok(())
+    }
+
+    /// Returns the length of this `Trytes`, in bytes.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use iota_trinary::*;
+    ///
+    /// let mut t = Trytes::try_from("ABC").unwrap();
+    ///
+    /// assert_eq!(t.len(), 3);
+    /// ```
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    /// Returns `true` if this `Trytes` has a length of zero, and `false` otherwise.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use iota_trinary::*;
+    ///
+    /// let mut t = Trytes::new();
+    /// assert!(t.is_empty());
+    ///
+    /// t.push('9');
+    /// assert!(!t.is_empty());
+    /// ```
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Splits the trytes into two at the given index.
+    ///
+    /// Returns a newly allocated `Trytes`. `self` contains bytes `[0, at)`, and
+    /// the returned `Trytes` contains bytes `[at, len)`.
+    ///
+    /// Note that the capacity of `self` does not change.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `at` is beyond the last code point of the trytes.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use iota_trinary::*;
+    ///
+    /// let mut t = Trytes::try_from("IOTA").unwrap();
+    /// let t2 = t.split_off(2);
+    /// assert_eq!(t.as_str(), "IO");
+    /// assert_eq!(t2.as_str(), "TA");
+    /// ```
+    pub fn split_off(&mut self, at: usize) -> Self {
+        let other = self.0.split_off(at);
+        Trytes(other)
+    }
+
+    /// Truncates this `Trytes`, removing all contents.
+    ///
+    /// While this means the `Trytes` will have a length of zero, it does not
+    /// touch its capacity.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use iota_trinary::*;
+    ///
+    /// let mut t = Trytes::try_from("FOO").unwrap();
+    ///
+    /// t.clear();
+    ///
+    /// assert!(t.is_empty());
+    /// assert_eq!(0, t.len());
+    /// ```
+    pub fn clear(&mut self) {
+        self.0.clear()
+    }
+
+    /// Encode a string literal as bytes to trytes.
+    ///
+    /// ```
+    /// use iota_trinary::*;
+    ///
+    /// let mut t = Trytes::new();
+    ///
+    /// t.encode("Lorem");
+    ///
+    /// assert_eq!("VBCDFDTCAD", t.as_str());
+    /// ```
+    pub fn encode(&mut self, plain: &str) {
```

Comment:
'Encode' implies a 1:1 mapping between an input and an output, but this appends to the buffer. Consider renaming this to `append_encoding` perhaps?